### ISSUE TITLE
Loss spikes in late training stages

### ISSUE THREAD

**Issue by diqihe666** [lucas-maes/le-wm#36]
Hi, thanks for the great work on LeWorldModel and for open-sourcing the code!

I am currently training the model and reproducing the results of PushT, but I've encountered an issue with training instability in the later stages. In my run, I am observing some loss spikes.

Modifications made to the original code/config:
I kept the codebase exactly as it is, with only the following two hyperparameter changes: Batch Size: Changed from 128 to 256; Learning Rate: Changed from 5e-5 to 2e-5

Reason for modifications:
When I initially tried training with the exact default hyperparameters (bs=128, lr=5e-5), the model eventually hit a NaN loss in the later stages of training. 
With the new hyperparameters, the NaN issue is gone, but as you can see in the attached screenshot, starting around step 30k, there are many loss spikes, especially in sigreg loss.
<img width="2553" height="1653" alt="Image" src="https://github.com/user-attachments/assets/b0bc40fb-f8c6-4a73-ae80-d29abaaa248d" />

Have you encountered similar NaN or loss spike issues during your internal experiments?
Any insights or suggestions would be greatly appreciated. Thank you!

**Comment by csh020219** [comment]
I have got the same issue, but changing trainer.precision from bf16 to 32 resolved this

**Comment by lucas-maes** [comment]
Hi! Thank you very much for sharing this insight! I think as @csh020219 suggests, float32 should help here, as precision could be a problem as you scale batch size? maybe? 

In any case, I assume a good hp tuning should mitigate the problem (gradient clipping, etc). 

Could be very interesting to investigate why it's happening!
